### PR TITLE
feat(entity): Entity state, queues

### PR DESCRIPTION
**Pull request by JakubG-git**

### CODE REVIEW

File: bdi_game/src/simulation/entity.rs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+use super::entity_type::SurvivorData;
+use super::entity_type::ZombieData;
+use super::grid::GridPoint;
+pub type Survivor = Entity<SurvivorData>;
+pub type Zombie = Entity<ZombieData>;
+
+pub struct Entity<EntityType>
+//survivor or zombie
+{
+    pub id: u64,
```
Comment (stabor705) [inline bdi_game/src/simulation/entity.rs]: Not needed for now

File: bdi_game/src/simulation/entity.rs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    pub position: GridPoint,
+    pub data: EntityType,
```
Comment (stabor705) [inline bdi_game/src/simulation/entity.rs]: data: EntityType?

File: bdi_game/src/simulation/entity.rs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    pub actions: EntityActionsQueue,
+    pub state: EntityState,
+    pub state_history: EntityStateHistory,
```
Comment (stabor705) [inline bdi_game/src/simulation/entity.rs]: Not needed for now

File: bdi_game/src/simulation/entity.rs [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    pub is_action_complete: bool,
```
Comment (stabor705) [inline bdi_game/src/simulation/entity.rs]: Not needed for now

File: bdi_game/src/simulation/entity_actions_queue.rs [hunk 5]
```diff
@@ -0,0 +1,15 @@
+use super::grid::GridPoint;
+use std::collections::VecDeque;
+pub enum EntityCommand {
```
Comment (stabor705) [inline bdi_game/src/simulation/entity_actions_queue.rs]: Rename to EntityAction

**Review by stabor705 — CHANGES_REQUESTED** [review]

File: bdi_game/src/simulation/entity.rs (hunk 1 above)
Comment (JakubG-git) [reply bdi_game/src/simulation/entity.rs]: Resolved

File: bdi_game/src/simulation/entity.rs (hunk 2 above)
Comment (JakubG-git) [reply bdi_game/src/simulation/entity.rs]: Resolved

File: bdi_game/src/simulation/entity_actions_queue.rs (hunk 5 above)
Comment (JakubG-git) [reply bdi_game/src/simulation/entity_actions_queue.rs]: Fixed

File: bdi_game/src/simulation/entity.rs (hunk 3 above)
Comment (JakubG-git) [reply bdi_game/src/simulation/entity.rs]: Resolved

**Pull request merged**